### PR TITLE
API additions required for windows linux hybrid clusters 

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
https://jira.coreos.com/browse/SDN-578

required by https://github.com/openshift/cluster-network-operator/pull/327

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@adambkaplan @dcbw PTAL

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -272,6 +272,14 @@ type OVNKubernetesConfig struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayExtensionConfig allows us to configure an additional overlay network .
+	// if not specified it is unused
+	HybridOverlayConfig ExtensionConfig `json:"hybridOverlayConfig,omitempty"`
```
Comment (adambkaplan) [inline operator/v1/types_network.go]: @JacobTanenbaum if this is optional (as indicated by `omitempty`), please do the following:

1. Add `// +optional` to godoc comment.
2. Make `ExtensionConfig` a pointer.

/cc @squeed

Comment (squeed) [reply operator/v1/types_network.go]: yup yup, very good point.

Comment (JacobTanenbaum) [reply operator/v1/types_network.go]: Updated PR

File: operator/v1/types_network.go [hunk 2]
```diff
@@ -272,6 +272,15 @@ type OVNKubernetesConfig struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayExtensionConfig allows us to configure an additional overlay network .
+	// if not specified it is unused
+	// +optional
+	HybridOverlayConfig *ExtensionConfig `json:"hybridOverlayConfig,omitempty"`
+}
+
+type ExtensionConfig struct {
```
[the remark this answers is not in the document]
Comment (squeed) [reply operator/v1/types_network.go]: Definitely - we have plans for both a few small knobs, as well as a few related features we're dreaming up.

File: operator/v1/types_network.go [hunk 3]
```diff
@@ -272,6 +272,15 @@ type OVNKubernetesConfig struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayExtensionConfig allows us to configure an additional overlay network .
+	// if not specified it is unused
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply operator/v1/types_network.go]: Done

**Comment by JacobTanenbaum** [comment]
@adambkaplan addressed your comments, is this okay to merge?

File: operator/v1/types_network.go [hunk 4]
```diff
@@ -272,6 +272,16 @@ type OVNKubernetesConfig struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayExtensionConfig configures an additional overlay network for peers that are
+	// not using OVN.
+	// If not specified it is unused
```
Comment (adambkaplan) [inline operator/v1/types_network.go]: @JacobTanenbaum is this statement necessary - imo an optional field implies "not used" unless a default is set elsewhere (and therefore is documented).

Comment (JacobTanenbaum) [reply operator/v1/types_network.go]: Done

**Comment by JacobTanenbaum** [comment]
@adambkaplan PTAL

**Comment by adambkaplan** [comment]
@JacobTanenbaum please squash, otherwise looks good

File: operator/v1/types_network.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (soltysh) [reply operator/v1/types_network.go]: I'd prefer this type having readable name than `ExtensionConfig`. Mabye `HybridOverlayConfig` or `HybridOverlayExtensionConfig`

**Comment by JacobTanenbaum** [comment]
@soltysh addressed your comment

**Pull request merged**